merlin-server/src/test/java/gov/nasa/jpl/aerie/merlin/server/services/TypescriptCodeGenerationServiceTest.java

```diff
@@ -1,17 +1,82 @@
 package gov.nasa.jpl.aerie.merlin.server.services;
 
+import gov.nasa.jpl.aerie.merlin.server.mocks.StubMissionModelService;
 import org.junit.jupiter.api.Test;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 class TypescriptCodeGenerationServiceTest {
 
   @Test
-  void testCodeGen() {
+  void testCodeGen() throws MissionModelService.NoSuchMissionModelException {
+    final var codeGenService = new TypescriptCodeGenerationService(new StubMissionModelService());
+
     assertEquals(
         """
             /** Start Codegen */
+            import * as AST from './constraints-ast.js';
+            import { Discrete, Real, Windows } from './constraints-edsl-fluent-api.js';
+            export type ActivityTypeName =
+              | "activity"
+              ;
+            export type ResourceName = "mode" | "state of charge";
+            export type DiscreteResourceSchema<R extends ResourceName> =
+              R extends "mode" ? ( | "Option1" | "Option2") :
+              R extends "state of charge" ? number :
+              void;
+            export function discreteResourceSchemaDummyValue<R extends ResourceName>(resource: R): DiscreteResourceSchema<R> {
+              return (
+                resource === "mode" ? "Option1" :
+                resource === "state of charge" ? 1.0 :
+                undefined
+              ) as DiscreteResourceSchema<R>;
+            }
+            export type RealResourceName = "state of charge";
+            export type ActivityParameters<A extends ActivityTypeName> =
+              A extends "activity" ? {Param: Discrete<string>, AnotherParam: Real, } :
+              void;
```

Comment:
I don't think `void` should be used here, but rather `undefined`. https://stackoverflow.com/a/58885486

Reply:
I thought `void` would act like a `never` type. Turns out there's an explicit `never` type. I'll use that instead. Thanks!

Reply:
We should probably throw at the end of the implementations then instead of returning undefined.

Reply:
Done. I added a `throw` if it comes out undefined, but I can't write a test for it because ideally that's unreachable.